### DISCUSSION TITLE
End of life: ubuntu 18.04 & python 3.7

### DISCUSSION THREAD

**Discussion by nim65s** [stack-of-tasks/pinocchio#1931] [Polls]
Hi,

Ubuntu 18.04 support is ending this month, and python 3.7 support is ending in 2 months. I'd like to plan to stop providing pip wheels for those two versions, as they are requiring more and more care:
- ubuntu 18.04 has GLIBC 2.27, which prevent us from upgrading build images to manylinux_2_28, and the old one is manylinux2014, a CentOS7 whose [GCC is bugged](https://bugzilla.redhat.com/show_bug.cgi?id=1546704) and impose us to use `-D_GLIBCXX_USE_CXX11_ABI=0`, which is [super boring](https://github.com/humanoid-path-planner/hpp-fcl/pull/386). But [18.04 support was requested not so long ago](https://github.com/stack-of-tasks/pinocchio/issues/1819).
- python 3.7 builds are not working anymore with `macos-latest` github images, so we have to explicitely specify `macos-10.15` only for python 3.7, and [`macos-10.15`  github images are deprecated](https://github.com/actions/runner-images).

But for now, we don't advertise any general rule for support, and dropping those without some prior discussions with the community might lead to even more required work. So here is a poll :)

Obviously, this is only about binary installations with pip, one can still compile from source or get binary installations from one of our other sources. source distribution through pip is also planed.

**Comment by jcarpent** [comment]
Thanks a lot @nim65s for this helpful poll
